shared/@hedvig-ui/TextDatePicker/index.tsx

```diff
@@ -0,0 +1,78 @@
+import styled from '@emotion/styled'
+import { Input, Shadowed } from '@hedvig-ui'
+import nlp from 'compromise'
+import dates from 'compromise-dates'
+import numbers from 'compromise-numbers'
+import formatDate from 'date-fns/format'
+import React, { HTMLAttributes } from 'react'
+import { Keys } from '../utils/key-press-hook'
+import { usePlatform } from '../utils/platform'
+
+export const getDate = (value: string) => {
+  nlp.extend(numbers)
+  nlp.extend(dates)
+
+  const date = nlp(value)
+    // @ts-ignore
+    .dates()
+    .get(0)
+
+  return date
+}
+
+const ChatTip = styled.div`
+  margin-top: 10px;
+  font-size: 0.8em;
+  color: ${({ theme }) => theme.semiStrongForeground};
+`
+
+interface TextDatePickerProps
+  extends Omit<HTMLAttributes<HTMLInputElement>, 'value'> {
+  value: string
+  setValue: any
+  setDate: any
```

Comment:
I think the props externally facing should be a date. Wheras the string is internal to the component
```suggestion
  value?: Date | null
  setValue: (date: Date) => void
```